dtm0/dtx.h

```diff
@@ -49,18 +49,28 @@ enum m0_dtm0_dtx_state {
 
 struct m0_dtm0_dtx {
 	/** An imprint of the ancient version of dtx.
-	 * This DTX was created at the begining of the time, and it was
-	 * propogated all over the codebase. Since it is really hard to
+	 * This DTX was created at the beginning of the time, and it was
+	 * propagated all over the codebase. Since it is really hard to
 	 * remove it, we instead venerate it by providing the very first
 	 * place in this new dtm0 structure. It helps to have a simple
 	 * typecast (without branded object) and no extra allocations for
-	 * this ancient but not really useful at this momemnt structure.
+	 * this ancient but not really useful at this moment structure.
 	 */
 	struct m0_dtx           dd_ancient_dtx;
 	/* see m0_dtm0_dtx_state */
 	struct m0_sm            dd_sm;
 	struct m0_dtm0_tx_desc  dd_txd;
 	struct m0_dtm0_service *dd_dtms;
+	uint32_t                dd_nr_executed;
+	struct m0_sm_ast        dd_exec_all_ast;
+
+	/* XXX: The implementation is very simple and it relies on the idea
+	 * of fully replicated requests. Therefore, all FOP submitted
```

Comment:
[* question *] Do we assert this anywhere in the path? may be in some invariant?